pkg/syndesis/action/attach.go

```diff
@@ -0,0 +1,161 @@
+package action
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	"github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/syndesis/configuration"
+	"github.com/syndesisio/syndesis-operator/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Attach a detached Syndesis installation to the Syndesis custom resource
+type Attach struct {}
+
+
+func (a *Attach) CanExecute(syndesis *v1alpha1.Syndesis) bool {
+	return syndesisInstallationStatusIs(syndesis, v1alpha1.SyndesisInstallationStatusAttaching)
+}
+
+func (a *Attach) Execute(syndesis *v1alpha1.Syndesis) error {
+
+	// Checking that there's only one installation to avoid stealing resources
+	if anotherInstallation, err := isAnotherActiveInstallationPresent(syndesis); err != nil {
+		return err
+	} else if anotherInstallation {
+		target := syndesis.DeepCopy()
+
+		target.Status.InstallationStatus = v1alpha1.SyndesisInstallationStatusNotInstalled
+		target.Status.Reason = v1alpha1.SyndesisStatusReasonDuplicate
+		target.Status.Description = "Cannot merge because another Syndesis resources is present in the same namespace"
+
+		logrus.Error("Cannot merge resource ", syndesis.Name, " because another Syndesis resources is present in the same namespace")
+		return sdk.Update(target)
+	}
+
+	logrus.Info("Attaching Syndesis installation to resource ", syndesis.Name)
+
+	for _, selector := range getAllManagerSelectors() {
+		for _, metaType := range getAllManagedResourceTypes() {
+
+			options := sdk.WithListOptions(&selector)
+			list := metav1.List{
+				TypeMeta: metaType,
```

Comment:
Is a type required for the list operation or is there a way to list all objects regardless of the kind which carry a certain label ?

Reply:
As said, will retry, but seems not possible...